pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +380,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.dockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
i know you didn't introduce it, but that's a horrible name for a field(way too generic) and the logic using it needs commenting. [/rant]